src/services/ledger/domain/entry-builder.ts

```diff
@@ -166,18 +176,29 @@ const EntryBuilderCreditWithUsdDebit = ({
     return entry
   }
 
-  const creditAccount = <D extends WalletCurrency>({
+  return {
+    creditLnd,
+  }
+}
+
+const EntryBuilderCreditWithUsdDebitLnd = ({
+  entry,
+  metadata,
+  debitAmount,
+  staticAccountIds: { dealerBtcAccountId, dealerUsdAccountId },
+}: EntryBuilderCreditState<"BTC">): EntryBuilderCreditWithUsdDebitLnd => {
+  const creditAccount = ({
     accountId,
     amount,
   }: {
     accountId: LedgerAccountId
-    amount: PaymentAmount<D>
+    amount: PaymentAmount<"USD">
   }) => {
-    entry.debit(dealerBtcAccountId, Number(amount.amount), {
+    entry.credit(dealerBtcAccountId, Number(debitAmount.amount), {
```

Comment:
Should we explore `debit`/`credit` wrappers where the `dealerBtcAccountId` lives in a `dealerBtcAccount` object that also has a `currency` property and there is some currency validation between the amount and ledger-account objects?

For e.g. it's still easy to pass a typed amount to the wrong account for the amount's currency at this step.

Reply:
Yeah this could certainly improve safety - not sure how this will get integrated but certainly an idea as we improve the outer layer of the ledger subdomain